### PR TITLE
BUILD.gn up to tests

### PR DESCRIPTION
**Pull request by Kangz**
PTAL at this new PR!

### CODE REVIEW

**Comment by SenorBlanco** [comment]
Looks good. I think we should land these as separate commits.

**Comment by Kangz** [comment]
Land which parts as separate commits?

**Comment by SenorBlanco** [comment]
I meant to land all four together, but unsquashed. Is that a bad idea? The D3D12 thing seems unrelated to the rest, for example.

**Comment by Kangz** [comment]
Oh, that's what I always do :P

File: scripts/dawn_overrides_with_defaults.gni [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# This files imports the overrides for Dawn but sets the defaults so that
+# projects including Dawn don't have to set dirs if they happen to use the
+# same.
+# It takes advantage of GN's variable scoping rules to define global variables
+# inside if constructs.
+
+import("//build_overrides/dawn.gni")
+
+if (!defined(dawn_spirv_cross_dir)) {
```
Comment (kainino0x) [inline scripts/dawn_overrides_with_defaults.gni]: I don't know how gn works, but it _looks_ like what would happen here is:

- importing `build_overrides/dawn.gni` sets `dawn_spirv_cross_dir`
- it's always defined here, so this `if` condition always fails

But I'm probably totally wrong.

Comment (Kangz) [reply scripts/dawn_overrides_with_defaults.gni]: If you're in a project whose `//build_overrides/dawn.gni` doesn't set the `dawn_spirv_cross_dir` then the `if` is taken. I actually checked it work by commenting out the import (and setting `dawn_standalone` that doesn't have defaults)

**Comment by Kangz** [comment]
Finally figured out you can click "requests" on Travis to see what requests it gets, and that it complained about .travis.yml not parsing.

**Pull request merged**